src/main/java/make_comment/use_case/MakeCommentInteractor.java

```diff
@@ -0,0 +1,102 @@
+package make_comment.use_case;
+
+
+import entities.Comment;
+import entities.CurrentUser;
+
+
+import java.util.*;
+
+/**
+ * The MakeCommentInteractor in the use case layer implements the MakeCommentInputBoundary interface.
+ * It takes in information from the outer layers of the architecture to run the use case by working directly
+ * with the entities and ultimately send information back out into the UI.
+ */
+public class MakeCommentInteractor implements MakeCommentInputBoundary {
+    private final MakeCommentGateway dataAccess;
+    private final MakeCommentOutputBoundary presenter;
+
+    public MakeCommentInteractor(MakeCommentGateway dataAccess, MakeCommentOutputBoundary presenter) {
+        this.dataAccess = dataAccess;
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void constructAndSaveCommentAndUpdatePost (MakeCommentRequestModel mCRM) {
+        try {
+            int userId = mCRM.getUserId();
+            String body = mCRM.getCommentBody();
+            int commentId = mCRM.getSelfId();
+            int postId = mCRM.getPostId();
+            constructAndSaveCommentHelper(userId, body, commentId);
+            updatePostHelper(postId, commentId);
+            MakeCommentResponseModel responseModel = new
+                    MakeCommentResponseModel(true, "");
+            this.presenter.present(responseModel);
+
+        } catch (Exception e) {
```

Comment:
We should be more specific with which Exception is being caught - eg maybe define our own